client/control/components/RecipeFilters.js

```diff
@@ -0,0 +1,83 @@
+import React, { PropTypes as pt } from 'react';
+import { connect } from 'react-redux';
+
+import ColumnMenu from 'control/components/ColumnMenu';
+
+import {
+  loadLocalColumns,
+} from 'control/actions/ColumnActions';
+
+/**
+ * Filters displayed above the RecipeList table.
+ */
+class RecipeFilters extends React.Component {
+  static propTypes = {
+    onSearchChange: pt.func.isRequired,
+    // connected
+    dispatch: pt.func.isRequired,
+    columns: pt.array.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {};
+  }
+
+  /**
+   * Loads user's last column display setup,
+   * as well as their last active list filters.
+   *
+   * @return {void}
+   */
+  componentWillMount() {
+    // load the last column setup user was viewing
+    this.props.dispatch(loadLocalColumns());
```

Comment:
I had seen all the actions go through `this.props.dispatch`, I have no problem switching it over though